### ISSUE TITLE
Use of enum and TimeZoneConvertible

### ISSUE THREAD

**Issue by Hout** [malcommac/SwiftDate#128]
The time zone naming is based on the official IANA time zone naming. NSTimeZone.knownTimeZoneNames produces the list of names. The type TimeZones captures this in an enum.

If we want to Swiftify the time zone naming, I would suggest to do it consistently:
- Use CamelCase instead of underlines.
- Follow the enum hierarchy of the slashes in the names. 

E.g. `America.Argentina.BuenosAires` instead of `America.Argentina_Buenos_Aires`.

**Comment by Hout** [comment]
In addition: what is the advantage of the current enum hierarchy compared to nested enums? Can't we just do:
```
enum TimeZone {
  enum America {
    case NewYork = "America/New_York"
    enum Argentina {
      case BuenosAires = "America/Argentina/Buenos_Aires"
      // etc
```

**Comment by malcommac** [comment]
Okay with both changes.
I don't understand your last remark: we have already nested enum; each enum is conform to a protocol that allows easy conversion to NSTImeZone so you can type Europe.Rome.toTimeZone() (or timeZone prop) and you don't need to make an extension of NSTImeZone which add an init with TimeZoneConvertible type to generate a new instance of NSTImeZone. Am I missing the point?

**Comment by Hout** [comment]
Nope I am missing the point ;-)
I am not that familiar with enums but I kind of cought up today.

**Comment by Hout** [comment]
Hmmm having looked at this for some time there are still two things where I think we can do better:
- We can set it up simpler: with just one big enum and without the protocol. Easier to understand and I can write a generator for the code (I am working on one for the locale anyway). 
- It does not work without the type name. I.e. we cannot put down `.Asia.Jakarta`; it should alway be `TimeZoneNames.Asia.Jakarta`.

Thoughts?

**Comment by Hout** [comment]
In the light of the msg above: what about consistent naming of the way we identify calendars, time zones and locales:
```enum CalendarName {
    case Gregorian // etc
enum TimeZoneName {
    case EuropeRome // etc
enum LocaleName {
    case EnglishGreatBrittain // etc
```

**Comment by Hout** [comment]
Check 68e121438f6934b1a15bf98410c2f85308cecea5

**Comment by Hout** [comment]
Fixed in v3.0.0
